app/domain/company.py

```diff
@@ -411,3 +412,51 @@ def find_employee_for_invitation(
     query = base_query.order_by(Email.address, Email.creation_time.desc())
 
     return query.yield_per(100).all()
+
+
+@log_execution
+def update_ceased_activity_status():
+    companies = (
+        Company.query.filter(Company.has_ceased_activity == False)
+        .order_by(nullsfirst(asc(Company.siren_api_info_last_update)))
+        .yield_per(100)
+    )
+
+    for company in companies:
+        (
+            has_ceased_activity,
+            siren_info,
+        ) = siren_api_client.has_company_ceased_activity(company.siren)
+        if has_ceased_activity:
+            app.logger.info(f"{company} has ceased activity...")
+
+            employments = Employment.query.filter(
+                Employment.company_id == company.id,
+                ~Employment.is_dismissed,
+                or_(Employment.end_date.is_(None)),
+                Employment.validation_status.in_(
+                    [
+                        EmploymentRequestValidationStatus.PENDING,
+                        EmploymentRequestValidationStatus.APPROVED,
+                    ]
+                ),
+            ).all()
+
+            app.logger.info(
+                f"#{len(employments)} employments will be terminated or dismissed"
+            )
+
+            for employment in employments:
+                if (
+                    employment.validation_status
+                    == EmploymentRequestValidationStatus.PENDING
+                ):
+                    employment.dismiss()
```

Comment:
suggestion(non-blocking): on pourrait indiquer un context
```suggestion
                    employment.dismiss(context="Entreprise cessée")
```

Reply:
Alors je trouvais que c'était une bonne idée mais je me suis rendu compte en essayant qu'en fait pour dismiss il faut un context_user. Sauf que l'on en a pas car c'est un cron qui va run le job...
Ici le dismiss était fait sur les employments pas encore acceptés (en fait je n'avais pas testé ce cas) donc je ne m'étais pas rendu compte que ça ne fonctionnait pas.
Je sais pas trop comment faire du coup.

Reply:
On peut simplement renseigner les colonnes dismissed_at et dismiss_context sans appeler la method .dismiss() qui vient de la classe `Dismissable` de `Employment` mais je trouve ça un peu naze. Le code pourrait faire l'hypothèse qu'un objet dismissed a forcément un dismiss_author à d'autres endroits. T'en penses quoi ?

Reply:
on a finalement décidé de supprimer les employment pending